pkg/cmd/server/certs/default_certs.go

```diff
@@ -0,0 +1,125 @@
+)
+
+const (
+	DefaultCADir = "ca"
+)
+
+type ClientCertInfo struct {
+	CertLocation configapi.CertInfo
+	SubDir       string
+	User         string
+	Groups       util.StringSet
+}
+
+func DefaultRootCAFile(certDir string) string {
+	return DefaultCertFilename(certDir, DefaultCADir)
+}
+
+func DefaultClientCerts(certDir string) []ClientCertInfo {
+	return []ClientCertInfo{
+		DefaultDeployerClientCertInfo(certDir),
+		DefaultOpenshiftLoopbackClientCertInfo(certDir),
+		DefaultKubeClientClientCertInfo(certDir),
+		DefaultClusterAdminClientCertInfo(certDir),
+	}
+}
+
+func DefaultDeployerClientCertInfo(certDir string) ClientCertInfo {
+	return ClientCertInfo{
+		CertLocation: configapi.CertInfo{
+			CertFile: DefaultCertFilename(certDir, "openshift-deployer"),
+			KeyFile:  DefaultKeyFilename(certDir, "openshift-deployer"),
+		},
+		SubDir: "openshift-deployer",
+		User:   "system:openshift-deployer",
+		Groups: util.NewStringSet("system:deployers"),
+	}
+}
+
+func DefaultOpenshiftLoopbackClientCertInfo(certDir string) ClientCertInfo {
+	return ClientCertInfo{
+		CertLocation: configapi.CertInfo{
+			CertFile: DefaultCertFilename(certDir, "openshift-client"),
+			KeyFile:  DefaultKeyFilename(certDir, "openshift-client"),
+		},
+		SubDir: "openshift-client",
+		User:   "system:openshift-client",
+	}
+}
+
+func DefaultKubeClientClientCertInfo(certDir string) ClientCertInfo {
+	return ClientCertInfo{
+		CertLocation: configapi.CertInfo{
+			CertFile: DefaultCertFilename(certDir, "kube-client"),
+			KeyFile:  DefaultKeyFilename(certDir, "kube-client"),
+		},
+		SubDir: "kube-client",
+		User:   "system:kube-client",
+	}
+}
+
+func DefaultClusterAdminClientCertInfo(certDir string) ClientCertInfo {
+	return ClientCertInfo{
+		CertLocation: configapi.CertInfo{
+			CertFile: DefaultCertFilename(certDir, "admin"),
+			KeyFile:  DefaultKeyFilename(certDir, "admin"),
+		},
+		SubDir: "admin",
+		User:   "system:admin",
+		Groups: util.NewStringSet("system:cluster-admins"),
+	}
+}
+
+func DefaultServerCerts(certDir string) []configapi.CertInfo {
+	return []configapi.CertInfo{
+		DefaultMasterServingCertInfo(certDir),
+		DefaultAssetServingCertInfo(certDir),
+	}
+}
+
+func DefaultMasterServingCertInfo(certDir string) configapi.CertInfo {
+	return configapi.CertInfo{
+		CertFile: DefaultCertFilename(certDir, "master"),
+		KeyFile:  DefaultKeyFilename(certDir, "master"),
+	}
+}
+
+func DefaultNodeServingCertInfo(certDir, nodeName string) configapi.CertInfo {
+	return configapi.CertInfo{
+		CertFile: DefaultCertFilename(certDir, "node_serving-"+nodeName),
+		KeyFile:  DefaultKeyFilename(certDir, "node_serving-"+nodeName),
+	}
+}
+
+func DefaultAssetServingCertInfo(certDir string) configapi.CertInfo {
+	return configapi.CertInfo{
+		CertFile: DefaultCertFilename(certDir, "master"),
+		KeyFile:  DefaultKeyFilename(certDir, "master"),
+	}
+}
+
+func DefaultCertDir(certDir, username string) string {
+	return path.Clean(path.Join(certDir, username))
```

Comment:
path.Join calls path.Clean internally (documented)... don't need to clean result